Church Locator pattern

sergsadovyi opened the pull request:
New Church Locator pattern added according to the provided layouts.

## Atoms
- `@atoms/icons/icon-close--thin` – thin variant of cross icon
- `@atoms/map/google-map` – basic Google Map component with markers and custom marker symbol

## Molecules
- `@molecules/church-locator/search` – search input for Locator
- `@molecules/church-locator/church-info` – church info popup for map markers
- `@molecules/church-locator/church-info-large` – church info popup for map markers with large info
- `@molecules/church-locator/search-results-item` – single search result block

## Organisms
- `@organisms/sections/locator-church-detail` – church detail component
- `@organisms/sections/locator-search-results` – search results list
- `@organisms/sections/locator-search-results-side` – search results list alternative view
- `@organisms/sections/locator-map-with-info` – map with basic popup opened
- `@organisms/sections/locator-map-with-large-info` – map with large popup opened

## Templates
- `@templates/locator-with-basic-info` – map with basic info popup
- `@templates/locator-with-large-info` – map with large info popup
- `@templates/locator-results-grid` – map with grid search results
- `@templates/locator-results-side` – map with side search results

## Pages
- `@pages/locator/locator-with-basic-info` – map with basic info popup
- `@pages/locator/locator-with-large-info` – map with large info popup
- `@pages/locator/locator-results-grid` – map with grid search results
- `@pages/locator/locator-results-side` – map with side search results

sergsadovyi committed:
feat: add Church Locator components and pages

sergsadovyi committed:
fix: map display & better image

sergsadovyi commented:
To finish the PR two things needed:
1. Map marker with a cross in better resolution (width ~50px, vector preferred)
2. Google Maps API key restricted to the localhost and alps.adventist.io domains.

@designerbrent

sergsadovyi committed:
feat: change the Google Maps API key to production

sergsadovyi committed:
fix(locator): html element typo

designerbrent commented on source/_patterns/01-molecules/church-locator/church-info.twig:
```diff
@@ -0,0 +1,9 @@
+<div class="c-church-info c-church-info--basic">
```
Add the class `can-be--dark-light`.

designerbrent commented on source/_patterns/01-molecules/church-locator/church-info.twig:
```diff
@@ -0,0 +1,9 @@
+<div class="c-church-info c-church-info--basic">
+  <div class="c-church-info__wrap">
```
Add the class `u-color--gray`

designerbrent commented on source/_patterns/01-molecules/church-locator/church-info.twig:
```diff
@@ -0,0 +1,9 @@
+<div class="c-church-info c-church-info--basic">
+  <div class="c-church-info__wrap">
+    <p class="c-church-info__title">{{ church_info.name }}</p>
```
Add the class `u-theme--color--darker`.

designerbrent commented on source/_patterns/01-molecules/church-locator/search-results-item.twig:
```diff
@@ -0,0 +1,7 @@
+<div class="c-search-results__item">
```
Add the class `can-be--dark-light`

designerbrent reviewed (changes requested):
I think this generally looks really good, but I have a few comments: When it's in dark mode, due to some of the styles, they are nearly unreadable. The classes above should help but probably won't take it all the way. 

## Dark mode for maps
What does it take to make the map show in dark mode? 

## Search box: 
When using this in the dark mode, the whole search box should be all white, since the map is still going to be white/light.

![2020-08-20 at 11 34 AM](https://user-images.githubusercontent.com/132864/90793362-a3ac9a80-e2d9-11ea-9101-f0e4849d88e3.png)

## Locator Results Side
On this template, the header is too big. Please fix this so it's the same size as the others.
![2020-08-20 at 11 40 AM](https://user-images.githubusercontent.com/132864/90793625-ef5f4400-e2d9-11ea-9de3-ac0ffc9679f5.png)

On the mobile/narrow view, the results should have a right margin.

![2020-08-20 at 11 38 AM](https://user-images.githubusercontent.com/132864/90793569-e1112800-e2d9-11ea-9cf6-1deac7bbfb15.png)

designerbrent commented:
You can find the *Dark mode* styles already available here: https://github.com/adventistchurch/alps/blob/v3.x/source/css/_settings.themes.scss#L819

We may need additional ones, and they can be added there.

designerbrent commented:
And the last item you requested: The vector pins. 

https://www.dropbox.com/s/kep6pz6h6n7rsh1/200820-ALPS-Pins.sketch?dl=0

Included are 2 versions and 2 sizes of each. The larger size would be intended to be used for the focused one?

sergsadovyi committed:
feat(locator): change the marker image and add focused variant

sergsadovyi committed:
feat(locator): set smaller font size for search results title

sergsadovyi committed:
feat(locator): adjust dark mode

sergsadovyi commented:
- added new marker icons
- added classes for dark mode (how are you checking the dark mode?)
- margin issue is not reproducible on my local, what browser version do you use?

@designerbrent

designerbrent commented:
@sergsadovyi ALPS Dark Mode can be tested at the bottom of any page: 

![2020-08-21 at 1 37 PM](https://user-images.githubusercontent.com/132864/90918888-7a603d00-e3b3-11ea-85db-e0795323a8b3.png)

designerbrent reviewed (changes requested):
☑️ Icon markers
- [ ] Please work make sure that when Dark Mode is enabled [☝️], the search box stays all white.
![2020-08-25 at 2 20 PM](https://user-images.githubusercontent.com/132864/91212467-413f0a00-e6de-11ea-9ded-598e19ceeea4.png)
- [ ] Margin issue: on the template `Locator Results Side`, please make sure the results have a side margin to separate it from the Sabbath column on the right. This can be reproduced in Safari, Firefox or Safari in a narrow view-port 
![2020-08-25 at 2 24 PM](https://user-images.githubusercontent.com/132864/91212865-cf1af500-e6de-11ea-93ec-c50ae94387c1.png)

designerbrent commented:
One more item: 

- [ ] On the template `Search Results Side`, the header is much too tall. 
![2020-08-25 at 2 32 PM](https://user-images.githubusercontent.com/132864/91213594-d5f63780-e6df-11ea-9ede-dc9843a006cb.png)

sergsadovyi committed:
fix(locator): search input with black theme

sergsadovyi committed:
fix(locator): add list item gap on Results Side

sergsadovyi commented:
Fixed Dark Mode and Margin issues.

Can't reproduce the Header bug. Please add more details: Browser, Resolution.

designerbrent reviewed (changes requested):
And this point, all my questions has been resolved except the header. I did a screen cast to show you what I'm seeing. 

[ALPS Locator Results Side Header bug](https://www.dropbox.com/s/ih2u2w6tpq39nw8/Locator_results_side_header_bug.mov?dl=0)

This is happening on Firefox, but I have seen the same behavior on Chrome and Safari. The problem seems to be with the grid: 
```
.c-map-main--2col {
    display: grid;
}
```

sergsadovyi committed:
fix(locator): header height on Results Side

designerbrent commented on source/css/_objects.locator.scss:
```diff
@@ -0,0 +1,385 @@
+      padding: 0 $space;
+      font-size: rem(20);
+      font-family: $ff-font--primary;
+      font-weight: bold;
```
The font styling here needs to be replaced with use of the already existing font class `u-font--primary--m`

designerbrent commented on source/css/_objects.locator.scss:
```diff
@@ -0,0 +1,385 @@
+    font-family: $ff-font;
+    font-weight: bold;
+    font-size: rem(10);
+    line-height: rem(14);
```
The font styling here needs to be replaced with use of the already existing font class `u-font--secondary--xs` and `u-text-strong `

designerbrent commented on source/css/_objects.locator.scss:
```diff
@@ -0,0 +1,385 @@
+  .c-church-info__location {
+    font-family: $ff-font;
+    font-size: rem(14);
+    line-height: rem(19);
```
The font styling here needs to be replaced with use of the already existing font class `u-font--secondary--s`

designerbrent commented on source/css/_objects.locator.scss:
```diff
@@ -0,0 +1,385 @@
+    font-family: $ff-font--primary;
+    font-weight: bold;
+    font-size: rem(20);
+    line-height: rem(24);
```
The font styling here needs to be replaced with use of the already existing font class `u-font--primary--m`

designerbrent replied:
and `u-text-strong `

designerbrent commented on source/css/_objects.locator.scss:
```diff
@@ -0,0 +1,385 @@
+      font-family: $ff-font;
+      font-weight: bold;
+      font-size: rem(10);
+      line-height: rem(14);
```
The font styling here needs to be replaced with use of the already existing font class `u-font--secondary--xs` and `u-text-strong`

designerbrent commented on source/css/_objects.locator.scss:
```diff
@@ -0,0 +1,385 @@
+    &__text {
+      font-family: $ff-font;
+      font-size: rem(14);
+      line-height: rem(18);
```
here too.

designerbrent commented on source/css/_objects.locator.scss:
```diff
@@ -0,0 +1,385 @@
+    border-left: 1px solid rgba(0, 0, 0, 0.8);
+    padding-left: rem(20);
+    font-family: $ff-font;
+    font-size: rem(26);
```
Please handle the sizing with `u-font--secondary--xl`.

designerbrent commented on source/css/_objects.locator.scss:
```diff
@@ -0,0 +1,385 @@
+    font-family: $ff-font--primary;
+    font-weight: bold;
+    font-size: rem(16);
+    line-height: rem(20);
```
Please review the font sizing and use existing font classes.

designerbrent commented on source/css/_objects.locator.scss:
```diff
@@ -0,0 +1,385 @@
+
+    @include media(">xlarge") {
+      font-size: rem(20);
+      line-height: rem(24);
```
Please review the font sizing and use existing font classes.

designerbrent commented on source/css/_objects.locator.scss:
```diff
@@ -0,0 +1,385 @@
+  &__location {
+    font-family: $ff-font;
+    font-size: rem(14);
+    line-height: rem(19);
```
Please review the font sizing and use existing font classes.

designerbrent commented on source/css/_objects.locator.scss:
```diff
@@ -0,0 +1,385 @@
+    font-family: $ff-font;
+    font-weight: bold;
+    font-size: rem(10);
+    line-height: rem(14);
```
Please review the font sizing and use existing font classes.

designerbrent commented on source/css/_objects.locator.scss:
```diff
@@ -0,0 +1,385 @@
+      font-weight: bold;
+      font-size: rem(10);
+      line-height: rem(14);
+      letter-spacing: 1px;
```
Please review the font sizing and use existing font classes.

designerbrent commented on source/css/_objects.locator.scss:
```diff
@@ -0,0 +1,385 @@
+    &__text {
+      font-family: $ff-font;
+      font-size: rem(14);
+      line-height: rem(18);
```
Please review the font sizing and use existing font classes.

designerbrent reviewed (changes requested):
Please review the current usage in this PR of the font styling (size, weight, text-transform) and use existing text styling classes. (https://styleguide.adventist.io/utilities/text)

Note: I'm OK with flexing a little up or down on the size in the comp and what you have defined here to match the existing style.

sergsadovyi committed:
fix(locator): replace font sizes with predefined

sergsadovyi commented:
* Replaced all `font-family` entries with predefined mixins.
* Adjusted line-height on marker tooltips text
* Prefer not to use `u-text-strong` in custom components. Displaying titles (and other elements) without strong is not implied by design. So `font-weight` should be defined by the element's class.

designerbrent reviewed (approved):
Looks good.

designerbrent commented:
Closes #484
